poet/poet.py

```diff
@@ -76,10 +85,106 @@ def walk(package):
     return sorted(discovered)
 
 
-def research_package(name, version=None):
+class PipSourceMetadataException(Exception):
+    pass
+
+@dataclass
+class PackageMetadata:
+    name: str
+    homepage: str
+    url: str
+    checksum: str
+    checksum_type: str
+
+def get_download_url_from_pip_source_file(module: str, pip_source_file: Path, output_dir=os.getenv("PIP_SOURCE_DIR")) -> str:
+    """
+    Returns the download URL for the pip source distribution.
+    This method will download the pip package from the source distribution. 
+
+    The standard out of this command contains an obfuscated URL and a regular URL that points to a .tar.gz file.
+
+    Args:
+        module (str): The name of the module to download.
+        output_dir (str): The directory to download the module to.
+
+    Returns:
+        str: The download URL for the pip source distribution.
```

Comment:
it looks like https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring will automatically generate docstrings by parsing the method signature?  i wonder whether it is smart enough to update generated docstrings when the signature changes, b/c this docstring is out of date now

Reply:
I looked at the docs for the plugin and saw that `Args` is no longer valid. Modified in [b8abb61](https://github.com/indigo-ag/homebrew-pypi-poet/pull/1/commits/b8abb6137f1b6d66ddc05e52473cc3c83aec98a9)